_develop/connect/use-with-clarity.md

```diff
@@ -33,34 +26,78 @@ Connect allows you to open the authenticator with parameters indicating the deta
+    icon: 'https://example.com/icon.png'
+  },
+  finished: (data) => {
+    console.log('TX ID:', data.txId);
+    console.log('Raw TX:', data.txRaw);
+  },
+};
+
+await openContractCall(opts);
+```
 
-When signing a transaction, you need to specify a few details. Here is the exact interface that describes what options you have:
+When calling `openContractCall`, you need to specify a few details. Here is the exact interface that describes what options you have:
 
 ```ts
-export interface ContractCallOptions {
+interface ContractCallOptions {
   contractAddress: string;
   functionName: string;
   contractName: string;
   functionArgs?: any[];
   authOrigin?: string;
-  userSession?: UserSession;
-  appDetails?: AuthOptions['appDetails'];
-  finished?: (data: FinishedTxData) => void;
+  appDetails: AuthOptions['appDetails'];
+  finished: (data: FinishedTxData) => void;
 }
 ```
 
-parameter | type | default | optional | description
----|---|---|---|---
-contractAddress | string | | false | The Stacks address that published this contract
-contractName | string | | false | The name that was used when publishing this contract
-functionName | string | | false | The name of the function you're calling. This needs to be a [public function](/core/smart/clarityRef.html#define-public).
-functionArgs | array | | false | The arguments you're calling the function with. You'll need to provide the Clarity type with each argument. See the below section for details.
-userSession | UserSession | | true | A `UserSession` instance
-appDetails | object | | false | A dictionary that includes `name` and `icon`
-finished | function | | false | A callback that is fired when the transaction is signed and broadcasted. Your callback will receive an object back with a `txId` and a `txRaw`, both of which are strings.
+parameter | type | optional | description
+---|---|---|---
+contractAddress | string | false | The Stacks address that published this contract
+contractName | string | false | The name that was used when publishing this contract
+functionName | string | false | The name of the function you're calling. This needs to be a [public function](/core/smart/clarityRef.html#define-public).
+functionArgs | array | false | The arguments you're calling the function with. You'll need to provide the Clarity type with each argument. See the below section for details. Defaults to `[]`.
+appDetails | object | false | A dictionary that includes `name` and `icon`
+finished | function | false | A callback that is fired when the transaction is signed and broadcasted. Your callback will receive an object back with a `txId` and a `txRaw`, both of which are strings.
+authOrigin | string | true | The location of the authenticator. This is only necessary when developing the authenticator locally, or when using beta features. Defaults to `"https://app.blockstack.org"`.
 
-#### Passing Clarity types with function arguments
+### Passing Clarity types with function arguments
```

Comment:
Could we link to this article where we mention principals? https://docs.blockstack.org/core/smart/principals.html